docs/dev/compiler-toolchain/overview.md

```diff
@@ -0,0 +1,66 @@
+# Overview
+
+This documentation introduces an LLVM-based compiler toolchain for smart contract languages with Ethereum Virtual Machine
+(EVM) support. The toolchain works on top of existing compilers and requires their output, which typically includes
+intermediate representations (IRs), abstract syntax trees (ASTs), and auxiliary contract metadata and documentation.
+At the moment, languages such as Solidity and Vyper are supported in this way.
+
+The toolchain roughly consists of the following parts:
+
+1. High-level source code compilers (`solc` and `vyper`)
+2. IR compilers, or LLVM front-ends (`zksolc` and `zkvyper`)
+3. [The LLVM framework](./llvm.md) with a zkEVM back-end, emitting the zkEVM text assembly
+4. The assembler, producing the zkEVM bytecode from the text assembly
+5. Hardhat plugins, setting up the environment
+
+## High-level Source Code Compilers
+
+The high-level source code compilers are the original upstream compilers that are called by the IR compilers as
+child processes to process their output. We use two such compilers at the moment of writing:
+
+- [solc](https://github.com/ethereum/solc-bin): the official Solidity compiler. [See the documentation](https://docs.soliditylang.org/en/latest/)
+- [vyper](https://github.com/vyperlang/vyper/releases): the official Vyper compiler [See the documentation](https://docs.vyperlang.org/en/latest/index.html)
+
+## IR Compilers
+
+Our toolchain includes LLVM front-ends that process the output of high-level source code compilers:
+
+- [zksolc](https://github.com/matter-labs/zksolc-bin); calling `solc` as a child process. [See the documentation](./solidity.md)
+- [zkvyper](https://github.com/matter-labs/zkvyper-bin); calling `vyper` as a child process. [See the documentation](./vyper.md)
+
+These IR compilers perform the following steps:
+
+1. Receive the input, which is usually a standard or combined JSON passed by the hardhat plugin via the standard input.
+2. Modify the input, save the relevant data, and pass the input down to the underlying high-level source code compiler, which is called as a child process.
+3. Receive the IRs and additional metadata from the underlying compiler, save the relevant data.
+4. Compile the IRs into LLVM IR, resolving dependencies and adding the zkEVM extra data to the output on the way.
+5. Provide the output accordingly to the input method the IR compiler is called with.
```

Comment:
We shouldn't drop this.
Now it looks like the output is provided to some input method, and it can imply that some actual method of some or interface is called with the output.
But the compiler provides the output **in the format that depends on the input method the user has chosen**.

Reply:
Honestly, without this adverb the sentence loses its sense to me.
The logic behind this is:
- if you're using the free-form I/O, the output will be in the free-form format
- if you're passing an input combined JSON, you'll get an output combined JSON
- if you're passing an input standard JSON, you'll get an output standard JSON

For the combined and standard JSON though, the underlying solc provides the template, so zksolc is not responsible for its correctness.
To be precise here, I suggest:
```
Provide the output matching the format of the input method the IR compiler is called with.
```